OneCommunityGlobal/HighestGoodNetworkApp
# Anirudh D - permissions management: Create constants for permission keys to replace passing strings directly into hasPermission()

# Description
Create constants for permission keys to replace passing strings directly into hasPermission()
![Screenshot 2024-03-12 at 5 05 08 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/1dbdcba1-869e-43fe-be23-ec9934777bc4)
The permissions were passed into `hasPermission()` as a hard-coded string in the frontend. Eg - `hasPermission('deleteWbs')`

To avoid manual error, I have created a separate constants file under `src/utils/constants.js` and I have created an object that has all the permissions being used. So the above example is changed to - `hasPermission(permissions.projects.deleteWbs)` and `constants.js` has the following entry - `deleteWbs: 'deleteWbs'`
![Screenshot 2024-03-12 at 5 10 22 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/490c310d-178f-4de1-b61c-3a13ace5c680)

## Related PRS (if any):
Use the latest backend
…

## Main changes explained:
Changed hard-coded permission strings to constants except in `src/components/PermissionsManagement/RolePermissions.jsx`

## How to test:
1. check into the current branch on the front end and the development branch on the backend
2. do `npm install` and `npm run start:local` to run this PR locally and do `npm install`, `npm run build && npm run start` on the development backend
3. Clear site data/cache
4. log in as any user and check if you encounter any bugs. Every user should function as the role it's assigned 

## Screenshots or videos of changes:
![Screenshot 2024-03-12 at 5 15 10 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/593e332c-ac5a-4720-b24f-e939e5b3fa46)
![Screenshot 2024-03-12 at 5 15 02 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/9b1da9e9-f9bc-4589-9484-6fef4a419bc5)
